table/internal_iterator.h

```diff
@@ -55,11 +60,12 @@ class InternalIteratorBase : public Cleanable {
   // REQUIRES: Valid()
   virtual void Next() = 0;
 
-  virtual bool NextAndGetResult(Slice* ret_key) {
+  virtual bool NextAndGetResult(IterateResult* result) {
     Next();
     bool is_valid = Valid();
     if (is_valid) {
-      *ret_key = key();
+      result->key = key();
+      result->may_be_out_of_upper_bound = MayBeOutOfUpperBound();
```

Comment:
How about set it to `true` here to avoid a virtual function if `NextAndGetResult()` is not explicitly implemented?